mixnet/client/src/sender.rs

```diff
@@ -89,15 +98,24 @@ impl<R: Rng> Sender<R> {
     }
 
     async fn send_packet(
+        cache: &ConnectionCache,
         packet: Box<SphinxPacket>,
         addr: NodeAddressBytes,
     ) -> Result<(), Box<dyn Error>> {
         let addr = SocketAddr::try_from(NymNodeRoutingAddress::try_from(addr)?)?;
         tracing::debug!("Sending a Sphinx packet to the node: {addr:?}");
 
-        let mut socket = TcpStream::connect(addr).await?;
-        let body = Body::new_sphinx(packet);
-        body.write(&mut socket).await?;
+        if let Some(stream) = cache.get(&addr) {
+            let mut stream = stream.lock().await;
+            let body = Body::new_sphinx(packet);
+            body.write(&mut *stream).await?;
+        } else {
+            let mut socket = TcpStream::connect(addr).await?;
+            let body = Body::new_sphinx(packet);
+            body.write(&mut socket).await?;
+            cache.insert(addr, Arc::new(Mutex::new(socket)));
+        }
```

Comment:
I think you already implemented kind of the batching that @Zeegomo mentioned.

p.s.
Even though this mechanism doesn't have buffering (that might reduce the number of calling `write`), I think it's okay. That's the different problem and that's not what Zeegomo suggested, if I understand correctly.